src/components/notification/NotificationBadge.tsx

```diff
@@ -0,0 +1,78 @@
+import { Intent, Popover, PopoverInteractionKind, Position, Tag } from '@blueprintjs/core';
+import * as React from 'react';
+import { AcademyNotification, AcademyNotificationType } from './notificationShape';
+
+type OwnProps = {
+  className?: string;
+  enableHover?: boolean; // enable or disable hover popover option
+  large?: boolean; // enable to use large style
+  notifications: AcademyNotification[];
+};
+
+export type DispatchProps = {
+  handleAcknowledgeNotification: (notificationIds: number[]) => void;
+};
+
+const NotificationBadge: React.SFC<OwnProps & DispatchProps> = props => {
+  if (!props.notifications.length) {
+    return null;
+  }
+
+  const makeNotificationTags = (notifications: AcademyNotification[]) => (
+    <div className="col">{notifications.map(makeNotificationTag)}</div>
+  );
+
+  const makeNotificationTag = (notification: AcademyNotification) => {
+    const onRemove = () => props.handleAcknowledgeNotification([notification.id]);
+
+    return (
+      <Tag
+        className="row"
+        style={{ backgroundColor: 'transparent', marginLeft: '0.4rem', marginRight: '0.4rem' }}
+        key={`${notification.id}`}
+        minimal={true}
+        multiline={true}
+        onRemove={onRemove}
+      >
+        {makeNotificationMessage(notification.type)}
+      </Tag>
+    );
+  };
+
+  return (
+    <Popover
+      className={props.className}
+      content={makeNotificationTags(props.notifications)}
```

Comment:
If the content is formed from an array of notifications that you map over, wont it have many messages stacked?

How will that look? Why not just have one message and the number?

Reply:
I think it's good to differentiate the notifications, hence we can have multiple notifications per notification badge. I limited the popover for assessments/submissions only as those will have at most 4 associated notifications.